src/repositories/repository.service.ts

```diff
@@ -12,10 +16,22 @@ export class RepositoryService {
   constructor(
     @InjectModel('Repository')
     private readonly repoModel: Model<RepositoryDocument>,
+    private issueService: IssueService,
+    private commitService: CommitService,
+    private releaseService: ReleaseService,
+    private pullRequestService: PullRequestService,
   ) {}
 
   public async initializeRepository(createRepoDto: CreateRepositoryDto) {
-    return this.getRepo(createRepoDto);
+    const repo = await this.getRepo(createRepoDto);
+    await this.issueService.storeIssues(createRepoDto, repo._id);
+    await this.commitService.storeCommits(createRepoDto, repo._id);
+    await this.releaseService.storeReleases(createRepoDto, repo._id);
+    await this.pullRequestService.storePullRequestDiffsForRepo(
+      createRepoDto,
+      repo._id,
+    );
+    return repo;
```

Comment:
`repo` is a `RepositoryDocument`, but instead of having the `storeX` methods expect a `RepositoryDocument` we "cast" it down to a `CreateRepositoryDto` and pass the `_id` as additional parameter.  
I think that's a bit weird.